### PR TITLE
feat: adjust component analysis for java-api

### PR DESCRIPTION
**Pull request by xieshenzh**
Get rid of the lsp. Create UI for component analysis that leverages exhort-java-api.

### CODE REVIEW

**Comment by xieshenzh** [comment]
@zvigrinberg @ilan-pinto Please take a look. Thanks.

File: src/main/resources/META-INF/plugin.xml [hunk 1]
```diff
@@ -1,10 +1,10 @@
 <idea-plugin>
-  <id>org.jboss.tools.intellij.analytics</id>
-  <name>Dependency Analytics</name>
-  <version>1.0</version>
-  <vendor email="developers@redhat.com" url="https://www.redhat.com">Red-Hat</vendor>
+    <id>org.jboss.tools.intellij.analytics</id>
+    <name>Dependency Analytics</name>
+    <version>1.0</version>
```
Comment (ilan-pinto) [inline src/main/resources/META-INF/plugin.xml:4]: should be version 0.7

**Review by ilan-pinto — CHANGES_REQUESTED** [review]

**Comment by zvigrinberg** [comment]
@xieshenzh Very good work!.
In addition to @ilan-pinto comment, kindly update README.md as well , to indicate that currently only npm and maven package managers are supported.

Did you had a chance to test it on couple of pom.xml and package.json manifests?
have you had a chance to test it on manifests with the exhortignore feature? ( checking the behavior that the plugin ignore them completely - the java-api shouldn't return vulnerabilities/recommendations of them in the report).

**Review by zvigrinberg — COMMENTED** [review]
@xieshenzh seems to be Very good work!.
In addition to @ilan-pinto comment, see also couple of minor comments of mine.
 kindly update README.md as well , to indicate that currently only npm and maven package managers are supported in this version.

Did you had a chance to test it on couple of pom.xml and package.json manifests?
have you had a chance to test it on manifests with the exhortignore feature? ( checking the behavior that the plugin ignore them completely - the java-api shouldn't return  vulnerabilities/recommendations of dependencies marked in manifests with exhortignore back in the report) - i think we can approve at this stage without this feature, but it would be nice to check whether it's working as expected.

File: src/main/resources/META-INF/plugin.xml (hunk 1 above)
Comment (xieshenzh) [reply src/main/resources/META-INF/plugin.xml:4]: Corrected it in this PR: https://github.com/redhat-developer/intellij-dependency-analytics/pull/129/files#diff-13198d14dd3abf13abd9cb906331ed7d9663fc081757b6845fd5f3c1850b5693

File: src/main/java/org/jboss/tools/intellij/componentanalysis/CAService.java [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+    }
+
+    public static void deleteReports(String filePath) {
+        deleteCache(filePath);
+    }
+
+    public static boolean performAnalysis(String packageManager,
+                                          String fileName,
+                                          String filePath,
+                                          Collection<Dependency> dependencys) {
```
[the remark this answers is not in the document]
Comment (xieshenzh) [reply src/main/java/org/jboss/tools/intellij/componentanalysis/CAService.java]: Renamed, thanks.

File: src/main/java/org/jboss/tools/intellij/componentanalysis/CAService.java [hunk 3]
```diff
@@ -0,0 +1,111 @@
...
+                        .expireAfterWrite(1, TimeUnit.MINUTES)
+                        .build());
+    }
+
+    private static void deleteCache(String filePath) {
+        getInstance().vulnerabilityCache.invalidate(filePath);
+    }
+
+    public static Map<Dependency, DependencyReport> getReports(String filePath, Collection<Dependency> dependencys) {
+        return getCache(filePath).getAllPresent(dependencys);
```
[the remark this answers is not in the document]
Comment (xieshenzh) [reply src/main/java/org/jboss/tools/intellij/componentanalysis/CAService.java]: Renamed.

**Comment by xieshenzh** [comment]
Thanks @zvigrinberg 

README.md is updated in this PR: https://github.com/redhat-developer/intellij-dependency-analytics/pull/129

I tested it mainly with these two projects: https://github.com/snyk-labs/java-goof and https://github.com/snyk-labs/nodejs-goof

Scanning npm dependencies works when I changed the java-api to run the `npm ls` command in the project directory that contains the `package.json` file. If the `package.json` file is copied to a temp folder, the `npm ls` command could not generate the correct versions of the dependencies. 
And `exhortignore` works for npm projects.

For maven projects, there is a scenario that the java-api doesn't work: the `mvn help:effective-pom` command fails for this [`pom.xml`](https://github.com/snyk-labs/java-goof/blob/main/todolist-goof/pom.xml), when it is copied to a temp folder.
And `exhortignore` doesn't work for maven projects, I think it is because of this [line](https://github.com/RHEcosystemAppEng/exhort-java-api/blob/f4d903b1cfa79fb29f2a89512dc4c751731692e8/src/main/java/com/redhat/exhort/providers/JavaMavenProvider.java#L153) of code. Also I am not sure if `exhortignore` works for maven dependencies with implicit versions (e.g. use a property as version or version is inherited).

I also need to check `exhortignore` on the client side. But I haven't implemented that yet.

**Comment by zvigrinberg** [comment]
> Thanks @zvigrinberg
> 
> README.md is updated in this PR: #129
> 
> I tested it mainly with these two projects: https://github.com/snyk-labs/java-goof and https://github.com/snyk-labs/nodejs-goof
> 
> Scanning npm dependencies works when I changed the java-api to run the `npm ls` command in the project directory that contains the `package.json` file. If the `package.json` file is copied to a temp folder, the `npm ls` command could not generate the correct versions of the dependencies. And `exhortignore` works for npm projects.
> 
> For maven projects, there is a scenario that the java-api doesn't work: the `mvn help:effective-pom` command fails for this [`pom.xml`](https://github.com/snyk-labs/java-goof/blob/main/todolist-goof/pom.xml), when it is copied to a temp folder. And `exhortignore` doesn't work for maven projects, I think it is because of this [line](https://github.com/RHEcosystemAppEng/exhort-java-api/blob/f4d903b1cfa79fb29f2a89512dc4c751731692e8/src/main/java/com/redhat/exhort/providers/JavaMavenProvider.java#L153) of code. Also I am not sure if `exhortignore` works for maven dependencies with implicit versions (e.g. use a property as version or version is inherited).
> 
> I also need to check `exhortignore` on the client side. But I haven't implemented that yet.

Thanks @xieshenzh 


For what concern the error you've experienced with mvn help:effective-pom, Yes you're right, when starting the API Clients, we probably didn't take into account pom.xml with modules inside it, as it was designed only to pass content of the pom.xml and that's it( for component analysis),  so in component analysis, we'll need to consider if we want to enhance it and to expose another overloaded method that will let you pass in a path/string of a directory containing pom.xml with directory modules - i'll track this activity by opening a new ticket.

for exhortignore with maven, both cases you mentioned, i'll check it out.

**Review by zvigrinberg — APPROVED** [review]

**Pull request merged**